File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        
        k = -1
        
        for i in range(0 , len(haystack)):
            
            if(haystack[i:len(needle) + i] == needle and (len(needle) + i) <= len(haystack) ):
                k = i
                break
        
        return k
            
        

File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.java
class Solution {
    public int strStr(String haystack, String needle) {
        Integer needle_length = needle.length();
        for(int i = 0 ; i < haystack.length() - needle_length + 1 ; ++i){
            if(needle.charAt(0) == haystack.charAt(i)){
                String cur_string = haystack.substring(i , i + needle_length);
                if(cur_string.equals(needle)) return i;
            }
        }
        return -1;
    }
}